src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/TempMultipleAlignmentsReclassifier.java

```diff
@@ -0,0 +1,211 @@
+            final List<AlignmentInterval> goodAlignments = goodAndBadAlignments._1;
+            final List<String> insertionMappings =
+                    goodAndBadAlignments._2.stream().map(AlignmentInterval::toPackedString).collect(Collectors.toList());
+            return new Tuple3<>(goodAlignments.size() < 2, // after fine tuning, a contig may have no good alignment left, or only 1
+                    new AlignedContig(tig.contigName, tig.contigSequence, goodAlignments, tig.hasEquallyGoodAlnConfigurations),
+                    insertionMappings);
+        });
+
+        // 1st class
+        final Tuple2<JavaRDD<Tuple3<Boolean, AlignedContig, List<String>>>, JavaRDD<Tuple3<Boolean, AlignedContig, List<String>>>> split =
+                RDDUtils.split(map, TempMultipleAlignmentsReclassifier::isNonInformative, false);
+        final JavaRDD<AlignedContig> garbage = split._1.map(Tuple3::_2);
+
+        // 3rd class
+        final Tuple2<JavaRDD<Tuple3<Boolean, AlignedContig, List<String>>>, JavaRDD<Tuple3<Boolean, AlignedContig, List<String>>>> split1 =
+                RDDUtils.split(split._2, TempMultipleAlignmentsReclassifier::hasOnly2Alignments, false);
+        final JavaPairRDD<AlignedContig, List<String>> twoAlignments =
+                split1._1.mapToPair(tuple3 -> new Tuple2<>(tuple3._2(), tuple3._3()));
+
+        // 2nd and 4th classes
+        final JavaPairRDD<AlignedContig, List<String>> multipleAlignments =
+                split1._2.mapToPair(tuple3 -> new Tuple2<>(tuple3._2(), tuple3._3()));
+        final JavaPairRDD<AlignedContig, List<String>> multipleAlignmentsIncompletePicture =
+                multipleAlignments.filter(pair -> hasIncomePicture(pair._1));
+        final JavaPairRDD<AlignedContig, List<String>> multipleAlignmentsCompletePicture =
+                multipleAlignments.filter(pair -> !hasIncomePicture(pair._1));
+
+        return new Tuple4<>(garbage, multipleAlignmentsIncompletePicture.keys(),
+                twoAlignments, multipleAlignmentsCompletePicture);
+    }
+
+    private static boolean hasIncomePicture(final AlignedContig contigWithMultipleAlignments) {
+        Utils.validateArg(contigWithMultipleAlignments.alignmentIntervals.size() > 2,
+                "assumption that input contig has more than 2 alignments is violated.\n" +
+                        contigWithMultipleAlignments.toString());
+
+        final AlignmentInterval head = contigWithMultipleAlignments.alignmentIntervals.get(0),
+                                tail = contigWithMultipleAlignments.alignmentIntervals.get(contigWithMultipleAlignments.alignmentIntervals.size()-1);
+
+        // tail not resuming the "direction of flow" started by head
+        if ( !head.referenceSpan.getContig().equals(tail.referenceSpan.getContig()) || head.forwardStrand != tail.forwardStrand)
+            return true;
+
+        // reference span anomaly
```

Comment:
Can you expand this comment more? Especially since the next block of code is based on a variable named `refSpanAnomaly`.. how does that differ from this check?